File: v1/src/main/java/com/api/v1/utils/RentOrderNumberGeneratorUtil.java
package com.api.v1.utils;

import java.math.BigInteger;
import java.time.ZonedDateTime;

public class RentOrderNumberGeneratorUtil {

    public static BigInteger generate() {
        int year = ZonedDateTime.now().getYear();
        String format = "%04d0000".formatted(year);
        BigInteger number = new BigInteger(format);
        number = number.add(BigInteger.ONE);
        return number;
    }

}

File: v2/src/main/kotlin/com/api/v2/rent/rent/RentOrderNumberGeneratorUtil.kt
package com.api.v2.rent.rent

import java.math.BigInteger
import java.time.ZonedDateTime

class RentOrderNumberGeneratorUtil {

    companion object {
        fun generate(): BigInteger {
            val year = ZonedDateTime.now().year
            val format = BigInteger("${year}0000")
            var number = format
            number = number.plus(BigInteger.ONE)
            return number
        }
    }

}